src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComponentLoweringPass.cs

```diff
@@ -130,38 +131,63 @@ public override void VisitTagHelperHtmlAttribute(TagHelperHtmlAttributeIntermedi
                 // node to the rather simpler form that property nodes usually have.
                 for (var i = 0; i < attribute.Children.Count; i++)
                 {
+                    IntermediateNode child = null;
```

Comment:
Would it be fair to say that the most complex aspect of the implementation is handling templates inside attributes? It looks like it takes up the bulk of the really complex cases you're having to handle here.

I'm wondering to what extent this scenario is going to come up in practice. Actually declaring any markup within an attribute looks pretty unfriendly. For example, given the choice between:

```
<MyFancyList GenericType="Customer" Data=customerRecords
   ItemTitle=@(@<div>@context.Name (@context.Email)</div>)>
   ItemBody=@(@<div>
       <ul>
       @foreach (var message in context.RecentMessages)
       {
           <li>@message</li>
       }
       </ul>
   </div>)>
</MyFancyList>
```

... and:

```
<MyFancyList GenericType="Customer" Data=customerRecords>
    <ItemTitle>@context.Name (@context.Email)</ItemTitle>
    <ItemBody>
       <ul>
       @foreach (var message in context.RecentMessages)
       {
           <li>@message</li>
       }
       </ul>
    </ItemBody>
</MyFancyList>
```

... people would choose the latter. Similarly for non-list scenarios, e.g., 

```
<MyDialog
    Header=@(@<div>Reorder @product.Name</div>)>
    Footer=@(@<div>
        <button onclick=@context.Accept>OK</button>
        <button onclick=@context.Dismiss>Close</button>
    </div>)>
    Some other markup here about @product
</MyDialog>
```

... versus:

```
<MyDialog>
    <Header>Reorder @product.Name</Header>
    <Footer>
        <button onclick=@context.Accept>OK</button>
        <button onclick=@context.Dismiss>Close</button>
    </Footer>
    <Body>
        Some other markup here about @product
    </Body>
</MyDialog>
```

Do you think there will be compelling scenarios for declaring templates inline in attributes? All this to say, let's chat about what those scenarios will be and whether the implementation will give us maintenance challenges as the templating feature is developed further in Razor itself.